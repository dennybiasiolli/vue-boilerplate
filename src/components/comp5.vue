<template>
  <div>
    <p>{{ message }}</p>
    <v-btn
      ref="btn"
      @click="reverseMessage">Reverse Message</v-btn>
  </div>
</template>

<script>
export default {
  data() {
    return {
      message: 'Hello Vue.js!'
    }
  },
  methods: {
    reverseMessage() {
      this.message = this.message.split('').reverse().join('')
    }
  }
}
</script>
