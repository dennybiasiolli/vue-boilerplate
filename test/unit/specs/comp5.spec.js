import { shallow } from 'vue-test-utils'
import comp5 from '@/comp5'

describe('comp5', () => {
  let component, vm, element

  beforeEach(() => {
    component = shallow(comp5)
    vm = component.vm
    element = component.element
  })

  test('should match snapshot', (done) => {
    expect(element).toMatchSnapshot()
    vm.reverseMessage()
    vm.$nextTick(() => {
      expect(element).toMatchSnapshot()
      done()
    })
  })

  // Evaluate the results of functions in
  // the raw component options
  test('sets the correct default data', () => {
    expect(typeof comp5.data).toBe('function')
    const defaultData = comp5.data()
    expect(defaultData.message).toBe('Hello Vue.js!')
  })

  // Inspect the component instance on mount
  test('correctly sets the seen flag when created', () => {
    expect(vm.message).toBe('Hello Vue.js!')
  })

  // Mount an instance and inspect the render output
  test('renders the correct message', () => {
    expect(element.textContent).toContain('Hello Vue.js!')
    const pElem = component.find('p').element
    expect(pElem.textContent).toBe('Hello Vue.js!')
    jest.spyOn(vm, 'reverseMessage')
    component.update() // Forces to re-render, applying changes on template
    component.find('button').trigger('click')
    expect(vm.reverseMessage).toHaveBeenCalled()
    expect(pElem.textContent).toBe('!sj.euV olleH')
    component.find('button').trigger('click')
    expect(pElem.textContent).toBe('Hello Vue.js!')
  })
})
