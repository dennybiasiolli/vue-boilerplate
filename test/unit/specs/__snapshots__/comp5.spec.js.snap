// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comp5 should match snapshot 1`] = `
<div>
  <p>
    Hello Vue.js!
  </p>
   
  <button>
    Reverse Message
  </button>
</div>
`;

exports[`comp5 should match snapshot 2`] = `
<div>
  <p>
    !sj.euV olleH
  </p>
   
  <button>
    Reverse Message
  </button>
</div>
`;
